### PR TITLE
Protect Seed input restore page from screenshots

### LINKED ISSUE #612: Protect Seed input restore page from screenshots
Currently it's possible to screenshot the seed word input screen on wallet restore. This input page needs to be protected from screenshots.

### LINKED ISSUE #613: Prevent dcrandroid window from showing a "preview" in task switcher
Since dcrandroid is a wallet app, the task switcher should not show the contents of the dcrandroid window, as a security safety measure. 

https://blog.jakelee.co.uk/how-to-disable-your-apps-preview-image-in-androids-task-app-switcher/

### PR DESCRIPTION
**Pull request by macsleven**
closes #612 
closes #613 
This PR includes:
- Protect Seed input restore page from screenshots
- Update codebase with dcrlibwallet master https://github.com/planetdecred/dcrlibwallet/pull/231
- Prevent app from showing a window preview in task switcher

### CODE REVIEW

File: app/src/main/java/com/dcrandroid/activities/RestoreWalletActivity.kt [hunk 1]
```diff
@@ -152,6 +154,12 @@ class RestoreWalletActivity : BaseActivity() {
         }.show(this)
     }
 
+    override fun onResume() {
+        super.onResume()
+        window.addFlags(WindowManager.LayoutParams.FLAG_SECURE)
+    }
+
+
```
Comment (JustinBeBoy) [inline app/src/main/java/com/dcrandroid/activities/RestoreWalletActivity.kt:162]: Remove line

File: app/src/main/java/com/dcrandroid/activities/privacy/AccountMixerActivity.kt [hunk 2]
```diff
@@ -37,7 +37,7 @@ class AccountMixerActivity : BaseActivity(), AccountMixerNotificationListener,
     override fun onResume() {
         super.onResume()
         multiWallet!!.removeTxAndBlockNotificationListener(this.javaClass.name)
-        multiWallet!!.addTxAndBlockNotificationListener(this, this.javaClass.name)
+        multiWallet!!.addTxAndBlockNotificationListener(this, true,this.javaClass.name)
```
Comment (JustinBeBoy) [inline app/src/main/java/com/dcrandroid/activities/privacy/AccountMixerActivity.kt:40]: please format it

**Review by JustinBeBoy — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 3]
```diff
@@ -7,14 +7,14 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 
 buildscript {
-    ext.kotlin_version = '1.5.21'
+    ext.kotlin_version = '1.6.10'
```
Comment (dreacot) [inline build.gradle:10]: this shouldn't be updated, at least not in this pr

File: build.gradle [hunk 4]
```diff
@@ -7,14 +7,14 @@
...
 
     repositories {
         mavenCentral()
         google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.2.2'
+        classpath 'com.android.tools.build:gradle:7.1.1'
```
Comment (dreacot) [inline build.gradle:17]: this shouldn't be updated, at least not in this pr

File: gradle/wrapper/gradle-wrapper.properties [hunk 5]
```diff
@@ -9,5 +9,5 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-all.zip
```
Comment (dreacot) [inline gradle/wrapper/gradle-wrapper.properties:12]: this shouldn't be updated, at least not in this pr

File: dcrlibwallet-ci-build.sh [hunk 6]
```diff
@@ -32,5 +32,6 @@ mkdir -p $GOPATH/src/github.com/planetdecred
 git clone https://github.com/planetdecred/dcrlibwallet $GOPATH/src/github.com/planetdecred/dcrlibwallet
 cd $GOPATH/src/github.com/planetdecred/dcrlibwallet
 export GO111MODULE=on && go mod vendor && export GO111MODULE=off
+go get golang.org/x/mobile/bind
```
Comment (dreacot) [inline dcrlibwallet-ci-build.sh:35]: the build script doesn't need to be updated

**Review by dreacot — CHANGES_REQUESTED** [review]

**Review by yaroyazeed — DISMISSED** [review]

**Review by yaroyazeed — APPROVED** [review]

**Pull request merged**